scripts/eqtl_hail_batch/conditional_analysis.py

```diff
@@ -0,0 +1,465 @@
+    significant_snps_df = adjusted_spearman_df
+
+    return significant_snps_df
+
+
+def merge_significant_snps_dfs(*df_list):
+    """
+    Merge list of list of sig_snps dataframes
+    """
+
+    merged_sig_snps = pd.concat(df_list)
+    pvalues = merged_sig_snps['p_value']
+    fdr_values = pd.DataFrame(list(multi.fdrcorrection(pvalues))).iloc[1]
+    merged_sig_snps = merged_sig_snps.assign(fdr=fdr_values)
+    merged_sig_snps['fdr'] = merged_sig_snps.fdr.astype(float)
+    merged_sig_snps.append(merged_sig_snps)
+
+    return merged_sig_snps
+
+
+def convert_dataframe_to_text(dataframe):
+    """
+    convert to string for writing
+    """
+    return dataframe.to_string()
+
+
+# Create click command line to enter dependency files
+@click.command()
+@click.option(
+    '--significant-snps', required=True, help='A space separated list of SNPs'
+)
+@click.option('--residuals', required=True, help='A CSV of gene residuals')
+@click.option(
+    '--keys',
+    required=True,
+    help='A TSV of sample ids to convert external to internal IDs',
+)
+@click.option(
+    '--output-prefix',
+    required=True,
+    help='A path prefix of where to output files, eg: gs://MyBucket/output-folder/',
+)
+@click.option(
+    '--iterations', type=int, default=4, help='Number of iterations to perform'
+)
+@click.option(
+    '--test-subset-genes',  # pylint: disable=too-many-locals
+    type=int,
+    help='Test with {test_subset_genes} genes, often = 5.',
+)
+def main(
+    significant_snps: str,
+    residuals,
+    keys,
+    output_prefix: str,
+    iterations=4,
+    test_subset_genes=None,
+):
+    """
+    Creates a Hail Batch pipeline for calculating EQTL using {iterations} iterations,
+    scattered across the number of genes. Note, {iterations} iterations are run, however
+    iterations start at 2, since round 1 is completed in `generate_eqtl_spearan.py`.
+    """
+    dataset = os.getenv('CPG_DATASET')
+    backend = hb.ServiceBackend(billing_project=dataset, remote_tmpdir=remote_tmpdir())
+    batch = hb.Batch(name='eQTL', backend=backend, default_python_image=DRIVER_IMAGE)
+
+    residual_df = pd.read_csv(AnyPath(residuals))
+    significant_snps_df = pd.read_csv(AnyPath(
+        significant_snps), sep=' ', skipinitialspace=True
+    )
+    
+    if test_subset_genes:
+        n_genes = test_subset_genes
+    else:
+        n_genes = test_subset_genes or get_number_of_scatters(
+            residual_df, significant_snps_df
+        )
+
+    filter_mt_job = batch.new_python_job('filter_mt')
+    copy_common_env(filter_mt_job)
+    filtered_mt_path = filter_mt_job.call(
+        prepare_genotype_info, keys_path=keys
+    )
+
+    previous_sig_snps_result = significant_snps_df  # pylint: disable=invalid-name
+    previous_residual_result = residual_df  # pylint: disable=invalid-name
+    # Perform conditional analysis for n iterations (specified in click interface)
+    for iteration in range(iterations):
```

Comment:
You can alter the range of `range`, so you don't need to manually edit the `iterations` each time
```suggestion
    for iteration in range(2, iterations +2):
```

Reply:
great tip, thanks! Will add that in and update the rest